### ISSUE TITLE
iOS app crashes on device in release build (IPA)

### ISSUE THREAD

**Issue by markuspalme** [dotnet/runtime#75758]
### Description

A very basic iOS app crashes as soon as a Nuget package reference is added for the library `MvvmCross` with the error:

> error: Failed to load AOT module 'System.Private.CoreLib' while running in aot-only mode: doesn't match assembly.

The package does not have any obvious dependencies that could cause a version conflict:

<img width="414" alt="image" src="https://user-images.githubusercontent.com/1481801/190672728-648fb294-cb49-473d-96c0-2b5460590207.png">

### Reproduction Steps

1. Create a new app with `dotnet new ios`
2. Set a valid bundle identifier in `Info.plist`
3. Add a reference to `MvvmCross`, version 9.0.1 or 9.0.0
4. Build an IPA file: `dotnet publish testapp.csproj -f:net6.0-ios -c:Release /p:CodesignKey="Apple Development: Markus Palme (XXXXXX)" /p:CodesignProvision="Test app" /p:ArchiveOnBuild=true -r ios-arm64`
5. Install the IPA file on an iPhone, start the app 
6. The launch screen shows quicky, then the app crashes with this message being visible in the console:

   <img width="1263" alt="image" src="https://user-images.githubusercontent.com/1481801/190670437-280dbba7-da99-497b-a10d-ef29683ed80e.png">

   > error: Failed to load AOT module 'System.Private.CoreLib' while running in aot-only mode: doesn't match assembly.
 
This is the full project file:

```<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0-ios</TargetFramework>
    <OutputType>Exe</OutputType>
    <Nullable>enable</Nullable>
    <ImplicitUsings>true</ImplicitUsings>
    <SupportedOSPlatformVersion>13.0</SupportedOSPlatformVersion>
    <MtouchLink>None</MtouchLink>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="MvvmCross" Version="9.0.1" />
  </ItemGroup>
</Project>
```

The linker settings don't seem to matter - I have tried without a setting in the project file, `SdkOnly` and `None`.

### Expected behavior

The app works on a physical device the same way it works inside the Simulator.

### Actual behavior

The app crashes on startup after showing the launch screen very briefly.

This error is visible in the device console:

<img width="1261" alt="image" src="https://user-images.githubusercontent.com/1481801/190673276-fd1aac54-c365-46b5-b655-6210e979b608.png">

### Configuration

dotnet: 6.0.401
iOS: 16.0
Architecture: ios-arm64

**Comment by markuspalme** [comment]
I don't really believe this a bug in the MvvmCross library as it's code is not even called but I have also opened a bug in their repo: https://github.com/MvvmCross/MvvmCross/issues/4484

**Comment by steveisok** [comment]
@mkhamoyan can you modify a runtime ios sample and see if you get the same behavior?

**Comment by markuspalme** [comment]
I have played around with this again, today. I found out that if I delete the `bin` and `obj` folders before the build it works and the app does not crash any more.

When you publish the app right after creating the project, then add the package reference and build again without the manual cleanup, you get the crash.

This looks like a bug in the build tools. I would assume that a `dotnet publish`would take care of cleaning up its cache itself.
